### PR TITLE
vendor: build scss folder with bundled variables and mixins

### PR DESCRIPTION
**Pull request by sharvit**
## PR Type

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (whitespace, formatting, missing semicolons, etc.)
* [ ] Refactoring (no functional changes, no api changes)
* [x] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Other… Please describe:

## Description

Change the hardcoded `./scss/variables.scss` and `./scss/mixins.scss` to be produced by a build process.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Checklist:

* [x] My change requires a change to the documentation.
* [x] I have updated the documentation accordingly.
* [x] I have read the [`contributing.md`](https://github.com/theforeman/foreman-js/blob/master/contributing.md).

### CODE REVIEW

**Comment by sharvit** [comment]
There are 2 commits because one is removing the current `scss` folder and the other put it under `.gitignore`.
from my experience, it never works in one commit.

File: packages/vendor-core/scss/variables.scss [hunk 1]
```diff
@@ -1,5 +1,9 @@
 /**
-  Third Party / Dependency imports here
+  Third Party variables imports here
 */
+$font-path: '~patternfly/dist/fonts/';
+$img-path: '~patternfly/dist/img/';
+$icon-font-path: '~patternfly/dist/fonts/';
```
Comment (tbrisker) [inline packages/vendor-core/scss/variables.scss]: IIUC this is overriding the variables from patternfly, shouldn't it be done after importing them?

Comment (sharvit) [reply packages/vendor-core/scss/variables.scss]: It should be done before because patternfly uses the scss `!default` feature.

Comment (sharvit) [reply packages/vendor-core/scss/variables.scss]: see https://sass-lang.com/documentation/variables#default-values

Comment (tbrisker) [reply packages/vendor-core/scss/variables.scss]: nice, learned something new

Comment (sharvit) [reply packages/vendor-core/scss/variables.scss]: btw it is done so solve this issue:
```scss
// lib.scss
$color: yellow;
$border: 1px solid $color;

// theme
@import 'lib';
$color: yellow;
```

will not affect the $border in this case

**Pull request merged**